### PR TITLE
Fix #731

### LINKED ISSUE #731: Do you have a plan to update package info of Godeps.json
I want to use the gin with the gRPC to backend communication.
But gRPC require newer versions of golang.org/x/net and protocolbuffer.
So do you have a plan to update package info of Godeps.json.

### PR DESCRIPTION
**Pull request by javierprovecho**
- Migrate from godep to govendor
- Update vendor.json
  - **We must** search or develop an alternative to github.com/mattn/go-isatty, it pulls golang.org/x/sys/unix which weights about 4 MB only for this file: https://github.com/mattn/go-isatty/blob/master/isatty_solaris.go
- Update .travis.yml
  - If we start to use heavily vendor systems, we must test against Go versions that support it, I mean `>= 1.6`.
  - App Engine is actually on `1.6.3`, waiting for security patch to be deployed.
  - Quicker CI with `git: depth 3`
- Packages refs updates here: https://github.com/gin-gonic/gin/pull/756/commits/44529e4a348c1d76594237677cec72eaa2753beb

### CODE REVIEW

**Comment by javierprovecho** [comment]
Packages refs updates here: https://github.com/gin-gonic/gin/pull/756/commits/c6dc464ab48cd14babb3541170590cd75707712f

**Comment by appleboy** [comment]
LGTM

File: .travis.yml [hunk 1]
```diff
@@ -1,12 +1,17 @@
 language: go
 sudo: false
 go:
-  - 1.4
-  - 1.5.4
```
Comment (tboerger) [inline .travis.yml]: You can keep this version if you set the required environment variable.

Comment (appleboy) [inline .travis.yml]: ping @tboerger.

Comment (tboerger) [inline .travis.yml]: Ok for me to drop it

**Pull request merged**